### ISSUE TITLE
Possible context memory leak?

### ISSUE THREAD

**Issue by validkeys** [hapijs/lab#868]
Hi Guys,

I was attempting to debug a growing memory leak in my application and think I know where it is. Currently, I use the beforeEach method to create one or more sequelize models and persist them to by DB. To ensure I can reference these models from each test, i do essentially the following:

```
beforeEach(async ({context}) => {
  let cache = {}
  cache.user = await Factory.create('user')
  context.cache = cache
})
```

After a while I noticed that none of these models were being garbage collected after running many tests. 

In runner.js, each test is being pushed into the reporters array:

```
state.report.tests.push(test);
```

...And therefore holding on to all of the variables I created in my context until all tests have run. Since the flags given to each test is a clone, I can never nullify the content of my original test. So as each of my tests get pushed into the tests array above, memory usage grows. 

I can't nullify that context in my afterEach hooks because I'm not getting the original context:

```
const flags = { notes: [], context: item.context || state.currentContext };
```

Perhaps context was not meant to be used for large objects such as data models, but I would think that once a test is run, it's context is no longer necessary. Unless I'm missing something, I'm happy to push a PR that nullifies the test's context once it's been run. 

Let me know. Thanks!

**Comment by geek** [comment]
Deleting the context from the test after the test is complete should be safe and I'll accept a PR to fix this. If you don't get around to it, I'll try to fix this up within the next week. Thanks for reporting this bug.

**Comment by validkeys** [comment]
@geek I've created a PR here to address the above concerns: https://github.com/hapijs/lab/pull/869

**Comment by geek** [comment]
@validkeys This is now published as 17.0.2, thanks for the help

**Comment by validkeys** [comment]
You got it. Thanks for the quick response time!
